lib/rule-list.js

```diff
@@ -155,17 +155,187 @@ export const SIMPLIFY_FRACTION = defineRuleString(
+            } else if (isPolynomialTerm(term)) {
+                newTerm = query.isNeg(term)
+                ? build.div(term.args[0], build.number(1))
+                : build.div(term, build.number(1))
+            } else {
+                newTerm = query.isNeg(term)
+                    ? term.args[0]
+                    : term
+            }
+            return newTerm
+        })
+
+        const result = build.add(
+            ...terms.map((term, i) => negatives[i]
+                         ? build.neg(term, {wasMinus: true})
+                         : term))
+        return result
+    }
+)
+
+// Finds common denominator in any scenario
+
+// e.g. 2/6 + 1/4 -> (2 * 2) / (6 * 2) + (1 * 3) / (4 * 3)
+// e.g. 2/(3 + x) + 2/3 -> (2 * 3)/ (3 * (3 + x)) + 2 * (3 + x) / (3 * (3 + x))
+// e.g. 3 + 2/3 -> (3 * 3) / 3 + 2 / 3
+export const COMMON_DENOMINATOR = defineRule(
+    (node) => {
+        if (query.isAdd(node)) {
+            const terms = node.args
+            
+            // checks if at least one of the terms is a fraction
+            const hasFraction = terms.some(term => isFraction(term))
+ 
+            let sameDenom
+            // rule cannot apply if all of the terms are fractions
+            // and they all have the same denominator
+            if (terms.every(term => isFraction(term))) {
+                const denom = getDenominator(terms[0])
+
+                sameDenom = terms.every(term =>
+                                        print(getDenominator(term)) == print(denom))
+            }
+
+            return hasFraction && !sameDenom ? {node} : null
+        }
+    },
+
+    (node) => {
+        node = canApplyRule(CONVERT_TO_FRACTION, node)
+            ? applyRule(CONVERT_TO_FRACTION, node)
+            : node
+
+        let terms = node.args
+        // an array storing the index where fraction is negative
+        const negatives = getNegatives(terms)
+
+        // get numerators and denominators of all fractions
+        let nums = terms.map(term => getNumerator(term))
+        let denoms = terms.map(term => getDenominator(term))
+        let newDenoms
+        let newNumerators
+
+        /*
+          If all denoms are numbers, the new denom is
+          the [denom * (LCM / denom)] and the new numerators
+          is [num * (LCM / denom)]
+
+          Else the new denom is the product of all the terms
+          and the new numerator is [num * all other denoms]
+        */
+
+        if(denoms.every(denom => query.isNumber(denom))) {
+            denoms = denoms.map(denom => query.getValue(denom))
+            const LCM = lcm(...denoms)
+
+            // remove multiplication by 1
+            newDenoms = denoms.map(
+                denom => denom == 1 
+                    ? build.number(LCM / denom)
+                    : LCM / denom == 1
+                    ? build.number(denom)
+                    : build.mul(build.number(denom),
+                                build.number(LCM / denom)))
```

Comment:
These chained ternaries are really hard to understand.  Can you rewrite this using `if-else` statements?